language-server/src/main/kotlin/tools/samt/ls/SamtSemanticLookup.kt

```diff
@@ -4,7 +4,10 @@ import tools.samt.parser.*
 import tools.samt.semantic.*
 
 abstract class SamtSemanticLookup<TKey, TValue> protected constructor() {
-    protected fun analyze(fileNode: FileNode, samtPackage: Package) {
+    protected lateinit var userMetadata: UserMetadata
```

Comment:
Could it make sense to pass this in the constructor? Also maybe storing the `samtPackage` in the same way as well could make sense, but I'm not sure